index.js

```diff
@@ -573,8 +574,23 @@ class Raspi extends EventEmitter {
     throw new Error('reportDigitalPin is not yet implemented');
   }
 
-  pingRead() {
-    throw new Error('pingRead is not yet implemented');
+  pingRead(config, handler) {
+    const pulseOut = config.pulseOut;
+    const pin = config.pin;
+    const sonar = new Sonar(config.pin);
+    
+    const pinInstance = this[getPinInstance](this.normalize(pin));
+    if (pinInstance.mode != INPUT_MODE) {
+      this.pinMode(pin, INPUT_MODE);
+    }
+    
+    const interval = setInterval(() => {
+      sonar.read(handler);
+    }, pulseOut);
```

Comment:
This is incorrect—`pingRead` should only read once and emit the reading, take a look at how it's used: https://github.com/rwaldron/johnny-five/blob/master/lib/proximity.js#L192-L199

Note that particle-io only does a one time event: https://github.com/rwaldron/particle-io/blob/master/lib/particle.js#L558 and voodoospark doesn't register reporting for that pin when it receives a `PING_READ` command.

Same with Firmata.js: https://github.com/jgautier/firmata/blob/master/lib/firmata.js#L1437


`pingRead` is really just garbage and everyone should use HCSR04 sensors with an I2C backpack. Generally, once there are more than 1 of these `pingRead` operations, the process is likely being blocked _too_ much.